test/errors.test.js

```diff
@@ -0,0 +1,340 @@
+'use strict'
+
+// Test lib/errors.js
+
+const path = require('path')
+
+const tape = require('tape')
+
+const logging = require('../lib/logging')
+const { createAPMError, _moduleNameFromFrames } = require('../lib/errors')
+
+const log = logging.createLogger('off')
+
+// Lookup the property "str" (given in dot-notation) in the object "obj".
+// If the property isn't found, then `undefined` is returned.
+function dottedLookup (obj, str) {
+  var o = obj
+  var fields = str.split('.')
+  for (var i = 0; i < fields.length; i++) {
+    var field = fields[i]
+    if (!Object.prototype.hasOwnProperty.call(o, field)) {
+      return undefined
+    }
+    o = o[field]
+  }
+  return o
+}
+
+// Test processing of Error instances by `createAPMError`.
+tape.test('#createAPMError({ exception: ... })', function (suite) {
```

Comment:
REVIEW-NOTE: This reproduces the old test/parsers.js '#parseError()' tests (except the 'should be able to choose number of source context line per frame type' tests that I think are more test code than helpful to ensure an int is passed through correctly)